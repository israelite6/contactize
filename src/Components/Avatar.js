import React from 'react';
import PropTypes from 'prop-types';
import {View, StyleSheet, Image} from 'react-native';

/**
 *
 * @param {string} source - Avatar url
 */
export default function Avatar({source, ...props}) {
  return (
    <View style={styles.root}>
      {source && (
        <Image
          source={source ? {uri: source} : null}
          resizeMode="contain"
          style={styles.root}
        />
      )}
    </View>
  );
}

Avatar.propTypes = {
  source: PropTypes.string,
};

Avatar.defaultProps = {
  source: null,
};

const styles = StyleSheet.create({
  root: {
    width: 60,
    height: 60,
    borderRadius: 30,
  },
});
