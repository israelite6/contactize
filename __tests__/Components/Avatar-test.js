/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../../src/Components/Avatar';

it('Avatar component renders correctly', () => {
  const tree = renderer.create(<Avatar />).toJSON();
  expect(tree).toMatchSnapshot();
});
