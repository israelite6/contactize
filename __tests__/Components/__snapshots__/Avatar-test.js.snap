// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 30,
      "height": 60,
      "width": 60,
    }
  }
/>
`;
